parlai/core/torch_agent.py

```diff
@@ -183,6 +183,17 @@ def add_cmdline_args(cls, argparser):
                  'text and __p2__ in front of past labels when available or '
                  'past utterances generated by the model. these are added to '
                  'the dictionary during initialization.')
+        agent.add_argument(
+            '--use-reply', default='label',
+            choices = ['label', 'model'],
+            help='Which previous replies to use as history. If label, use '
+            'gold dataset replies. If model, use model\'s own replies.')
+        agent.add_argument(
+            '--add-p1-after-newln', type='bool', default=False,
```

Comment:
be good if we can generalize the speaker token to use the id as well if you want (every agent has an id)

Reply:
we'll need to make sure we add those ids to our dictionary